components/Nav/RightNav.jsx

```diff
@@ -0,0 +1,89 @@
+import Link from 'next/link';
+import styled from 'styled-components';
+
+/*This code is for the style of right navbar which is changed when screen size is mobile. Also the if statement inside style components is for underline transition for the nav options when the screen is not mobile size*/
+const Ul = styled.ul`
+  list-style: none;
+  display: flex;
+  flex-flow: row nowrap;
+  font-size: 0.5em;
+  li {    
+    padding-right: 3em;
+    padding-top: 1em;
+    cursor: pointer;    
+  }    
+  ${ function ({open}){
+    if (open==false) {
+      return (
+        `.hover-underline-animation {
+          display: block;
+          position: relative;    
+        }
+        
+        .hover-underline-animation:after {
+          content: '';
+          position: absolute;
+          width: 60%;
+          transform: scaleX(0);
+          height: 2px;
+          bottom: 0;
+          left: 0;
+          background-color: #fff;
+          transform-origin: bottom right;
+          transition: transform 1s ease-out;
+        } 
+        
+        .hover-underline-animation:hover:after {
+          transform: scaleX(1);
+          transform-origin: bottom left;
+        }               
+        `
+        
+      );
+    }else{
+      return ``;
+    }
+  }
+
+  }
+  @media (max-width: 768px) {
+    margin: 0px;    
+    flex-flow: column nowrap;
+    background-color: #0D2538;    
+    position: fixed;
+    background: rgba( 1, 1, 1, 0.60 );
+    box-shadow: 0 8px 32px 0 rgba( 31, 38, 135, 0.37 );
+    backdrop-filter: blur( 13.0px );
+    -webkit-backdrop-filter: blur( 13.0px );    
+    border: 1px solid rgba( 255, 255, 255, 0.18 );
+    transform: ${({ open }) => open ? 'translateX(0)' : 'translateX(100%)'};
+    top: 0;
+    right: 0;
+    height: 1000px;
+    width: 100%;
+    padding-top: 3.5rem;
+    transition: transform 0.3s ease-in-out;
+    li {
+      font-size: 1.25em;
+      padding-top: 2em;
+      color: #fff;      
+    }
+  }
+`;
+
+const RightNav = ({ open }) => {
+  return (
+    <div className='overlay overlay-contentpush open'>
+    <Ul open={open}>      
```

Comment:
- A bit of space needed between text and animation

Reply:
sure